### PR TITLE
RELATED: ONE-4639 Custom styles for Button Component

### PR DESCRIPTION
**Pull request by NestorEncinas**
Button able to accept different styles

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-kit/styles/scss/Button/_variables.scss [hunk 1]
```diff
@@ -65,3 +65,22 @@ $button-negative-hover-bg: darken($button-negative-bg, 10%);
 $button-negative-focus-shadow: transparentize(lighten($button-negative-bg, 10%), 0.4);
 $button-negative-active-gradient-top: darken($button-negative-bg, 20%);
 $button-negative-active-gradient-bottom: $button-negative-hover-bg;
+
+// border-radius
+$button-border-radius: var(--gd-button-border-radius, 3px);
+$button-border-radius-max: calc(max(85px, $gd-button-border-radius));
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/styles/scss/Button/_variables.scss:71]: where is this $gd-button-border-radius defined?

File: libs/sdk-ui-kit/styles/scss/Button/_variables.scss [hunk 2]
```diff
@@ -65,3 +65,22 @@ $button-negative-hover-bg: darken($button-negative-bg, 10%);
 $button-negative-focus-shadow: transparentize(lighten($button-negative-bg, 10%), 0.4);
 $button-negative-active-gradient-top: darken($button-negative-bg, 20%);
 $button-negative-active-gradient-bottom: $button-negative-hover-bg;
+
+// border-radius
+$button-border-radius: var(--gd-button-border-radius, 3px);
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/styles/scss/Button/_variables.scss:70]: what about units?

File: libs/sdk-ui-kit/styles/scss/button.scss [hunk 3]
```diff
@@ -595,14 +600,14 @@
 
     > .gd-button:first-child,
     > *:not(.gd-button):first-child .gd-button {
-        @include border-radius(3px 0 0 3px);
+        border-radius: $button-group-childs-border-radius;
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/styles/scss/button.scss:603]: this should not be single variable. It should be here like :
 $button-border-radius 0 0 $button-border-radius

File: libs/sdk-ui-kit/styles/scss/button.scss [hunk 4]
```diff
@@ -595,14 +600,14 @@
...
     }
 
     > .gd-button:last-child,
     > *:not(.gd-button):last-child .gd-button {
         margin-right: 0;
 
-        @include border-radius(0 3px 3px 0);
+        border-radius: $button-group-childs-border-radius;
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/styles/scss/button.scss:610]: this should not be single variable. It should be here like :
 0 $button-border-radius $button-border-radius 0

**Review by ivan-nejezchleb — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1008/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2594/ : SUCCESS in 20m 24s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1011/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2599/ : SUCCESS in 22m 15s

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1013/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/374/ : UNSTABLE in 34m 53s

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1019/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/376/ : UNSTABLE in 24m 58s

File: libs/sdk-ui-kit/styles/scss/Button/_variables.scss (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-kit/styles/scss/Button/_variables.scss:70]: will be added in ThemeLoader

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1022/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2610/ : SUCCESS in 10m 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1024/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2612/ : SUCCESS in 15m 01s

**Comment by NestorEncinas** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1032/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/378/ : SUCCESS in 17m 44s

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/388/ : SUCCESS in 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/645/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2622/ : SUCCESS in 11m 45s

**Pull request merged**